Allow apps to specify a minimum OS version (#2239)
beeware/briefcase: docs/reference/platforms/android/gradle.rst

    <uses-permission android:name="android.permission.HIGH_SAMPLING_RATE_SENSORS">

``target_sdk_version``

> ----------------------

Reviewer: Even though there isn't any equivalent on the other platforms, would it be better to call this `target_os_version` to indicate that both settings are of the same type?

Author: Since it's an Android-specific setting, I was leaning towards preserving the "Android-ness" of it; but I given the comments below about how it's interpreted, maybe it makes sense to stay consistent with the other Briefcase setting.

The SDK version that the app will target. This controls the APIs that are available for

> use in your application. This is *not* the Android version; it is the underlying SDK

Reviewer: Technically the `compileSdkVersion` controls the available APIs, while `targetSdkVersion` controls various [behavior and compatibility changes](https://developer.android.com/guide/topics/manifest/uses-sdk-element#target). The latter is usually more important, since in most cases your available APIs are constrained by the `minSdkVersion` anyway.

Author: I've clarified the text to match the behaviour of the two settings. I've also mentioned the explicit settings (both here, and in the iOS/macOS settings).

    <uses-permission android:name="android.permission.HIGH_SAMPLING_RATE_SENSORS">

``target_os_version``
----------------------

> The API level that the app will target. This controls the version of Android tools that
> are used to build your app (by setting the ``compileSdkVersion`` for your app), and the

Reviewer: The tools version is controlled by yet another setting (`buildToolsVersion`), but that shouldn't usually be customized because it's so tightly coupled to Gradle.

```suggestion
The API level that the app will target. This controls the version of the Android SDK that
is used to build your app (by setting the ``compileSdkVersion`` for your app), and the
```
